packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -0,0 +1,119 @@
+import type { PageRuntime } from '../../server/config-shared'
+import type { NextConfig } from '../../server/config-shared'
+import { tryToExtractExportedConstValue } from './extract-const-value'
+import { parseModule } from './parse-module'
+import { promises as fs } from 'fs'
+
+export interface PageStaticInfo {
+  runtime?: PageRuntime
+  ssg?: boolean
+  ssr?: boolean
+}
+
+/**
+ * For a given pageFilePath and nextConfig, if the config supports it, this
+ * function will read the file and return the runtime that should be used.
+ * It will look into the file content only if the page *requires* a runtime
+ * to be specified, that is, when gSSP or gSP is used.
+ * Related discussion: https://github.com/vercel/next.js/discussions/34179
+ */
+export async function getPageStaticInfo(params: {
+  nextConfig: Partial<NextConfig>
+  pageFilePath: string
+  isDev?: boolean
+}): Promise<PageStaticInfo> {
+  const { isDev, pageFilePath, nextConfig } = params
+
+  const fileContent = (await tryToReadFile(pageFilePath, !isDev)) || ''
+  if (/runtime|getStaticProps|getServerSideProps/.test(fileContent)) {
+    const swcAST = await parseModule(pageFilePath, fileContent)
+    const { ssg, ssr } = checkExports(swcAST)
+    const config = tryToExtractExportedConstValue(swcAST, 'config') || {}
```

Comment:
**suggestion(non-blocking):** why not supporting also `var` and `let` ?

So we could get rid of 
```js
    if (declaration.kind !== 'const') {
      continue
    }
``` in `extract-const-value.ts` and rename all the things as `extractExportedValue()` + `tryToExtractExportedValue()`

For context: I'll reckon _it makes no sense to export a mutable variable_. And personally, I always strive using immutable code as much as possible. However it does not feel it is the purpose of this code to convey good practices: let's not add unnecessary constraints to the users.